### PR TITLE
ACS-8417: Fix required by HxP APIs request headers

### PR DESCRIPTION
**Pull request by krdabrowski**
https://hyland.atlassian.net/browse/ACS-8417

### CODE REVIEW

File: common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java [hunk 1]
```diff
@@ -39,36 +43,63 @@
...
+    }
+
+    public Map.Entry<String, String> getAuthHeader(String providerId)
+    {
+        return Map.entry(AUTHORIZATION, getAuthHeaderValue(providerId));
+    }
+
+    protected Map.Entry<String, String> getHxpEnvironmentHeader()
+    {
+        return new AbstractMap.SimpleEntry<>(HXP_ENVIRONMENT_HEADER, ofNullable(authProperties.getProviders().get(HXP_AUTH_PROVIDER))
```
Comment (dsibilio) [inline common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java:89]:
```suggestion
        return new AbstractMap.SimpleEntry<>(
            HXP_ENVIRONMENT_HEADER,
            ofNullable(authProperties.getProviders().get(HXP_AUTH_PROVIDER))
                .map(AuthProperties.AuthProvider::getEnvironmentKey)
                .orElse(null)
        );
```

I'd suggest formatting this differently to make it easier to read.

**Review by dsibilio — APPROVED** [review]
LGTM. Great work!

File: common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java (hunk 1 above)
Comment (krdabrowski) [reply common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java:89]: Great suggestion, thanks Domenico!

**Review by damianujma — APPROVED** [review]

**Review by mpichura — COMMENTED** [review]
I have one doubt on allowing `null` value in the header, rest LGTM

**Review by aonych — APPROVED** [review]

File: common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java [hunk 2]
```diff
@@ -39,36 +43,65 @@
...
+        return Map.entry(AUTHORIZATION, getAuthHeaderValue(providerId));
+    }
+
+    protected Map.Entry<String, String> getHxpEnvironmentHeader()
+    {
+        return new AbstractMap.SimpleEntry<>(
+                HXP_ENVIRONMENT_HEADER,
+                ofNullable(authProperties.getProviders().get(HXP_AUTH_PROVIDER))
+                        .map(AuthProperties.AuthProvider::getEnvironmentKey)
+                        .orElse(null));
```
[the remark this answers is not in the document]
Comment (krdabrowski) [reply common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java:93]: Yes, this is intended - this allows to call this method within `clearAuthHeaders()` and access the header key

[the remark this answers is not in the document]
Comment (krdabrowski) [reply common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/AuthService.java:98]: I've added a check to filter out possible null headers before applying them to request

**Pull request merged**